pages/index.tsx

```diff
@@ -199,17 +200,43 @@ const Home: NextPage = () => {
           Get started by editing <code className={classes.code}>pages/index.js</code>
         </p>
 
-        <BlogSection posts={FEATURED_DUMMY_DATA} variation="featured" />
-        <BlogSection title="Most Popular" posts={MOST_POPULAR_DUMMY_DATA} variation="grid" />
-        <BlogSection
-          title="Browse All"
-          posts={BROWSE_ALL_DUMMY_DATA}
-          variation="grid"
-          dropdown={true}
-        />
+        <BlogSection posts={posts} variation="featured" />
+        <BlogSection title="Most Popular" posts={posts} variation="grid" />
+        <BlogSection title="Browse All" posts={posts} variation="grid" dropdown={true} />
       </main>
     </div>
   );
 };
 
+export async function getStaticProps() {
+  const response = await fetchEntries();
+  const posts = await transformContentfulPosts(response);
+
+  return {
+    props: {
+      posts,
+    },
+  };
+}
+
+const transformContentfulPosts = (items: object[] | undefined): IBlogPost[] | void => {
+  if (!items) {
+    return;
+  }
```

Comment:
Maybe you can instead return an empty array in the case where the response is empty. That simplifies the types as well. Something like this:

```suggestion
const transformContentfulPosts = (items: object[] | undefined): IBlogPost[] => {
  if (!items) {
    return [];
  }
```